handler/resource.go

```diff
@@ -292,3 +292,50 @@ func (h *Handler) RecommendResource(w http.ResponseWriter, r *http.Request) {
 	}
 	return
 }
+
+// GetResource get a resource
+func (h *Handler) GetResource(w http.ResponseWriter, r *http.Request) {
+	resourceId, _ := strconv.ParseInt(mux.Vars(r)["resourceId"], 10, 64)
+	if resourceId == 0 {
```

Comment:
@tomipaul What if the resource Id is a string or nil

Reply:
@mikey2020 it won't even match at all if it is a random string or nil
This is because of the regex match in the route declaration

Reply:
@mikey2020 Yes, perhaps there would still be a `get all resources` endpoint but right now there isn't. There is a difference between a URL that does not match any route and a URL that matches but with invalid parameters. The former is the case here, any unmatched url gives a 405. Unless I remove the regex match from the route so that it becomes the latter, we can't specifically handle with message `please input a resource ID`. We might look into putting a catch all route though. Nevertheless, I don't think it's our call to point out a mistake in an invalid URL to a user rather we should be pointing out only if the URL matches but with invalid payload.